soa-rs-derive/src/fields.rs

```diff
@@ -117,11 +117,13 @@ pub fn fields_struct(
 
             fn from_slice(slice: &::soa_rs::Slice<Self::Item>) -> &Self {
                 // SAFETY: Self is `repr(transparent)` of Slice
+                #[allow(clippy::transmute_ptr_to_ptr)]
```

Comment:
Can we go with the safe code you mentioned in the PR? It would be nice to remove unsafe where possible.

Reply:
actually its still unsafe. clippy suggest to rewrite it like that:
```rust
fn from_slice(slice: &::soa_rs::Slice<Self::Item>) -> &Self {
    unsafe { &*(slice as *const _ as *const Self) }
}
```

Reply:
Ah, gotcha gotcha. I'll merge as-is then.